tests/write_float/main.cpp

```diff
@@ -17,7 +17,12 @@ inline void cudaCheck(libreCudaStatus_t error, const char *file, int line) {
 };
 #define CUDA_CHECK(err) (cudaCheck(err, __FILE__, __LINE__))
 
-int main() {
+int main(int argc, char *argv[]) {
+    if(argc != 2){
```

Comment:
This is well intentioned, but probably counterproductive. Tests should be standalone and launchable without arguments.
The .cubin file should probably be copied to the binary directory and the test should assert to find it in its working directory.